modules/storage-filesystem/snapshot.go

```diff
@@ -42,54 +42,71 @@ func (m *StorageFileSystemModule) StoreSnapshot(ctx context.Context, snapshot *s
 		}
 	}
 
-	if err := m.saveMeta(dstSnapshotPath, snapshot); err != nil {
+	if err := m.saveMeta(snapshot); err != nil {
 		return err
 	}
 
 	return nil
 }
 
-// TODO handle className
-func (m *StorageFileSystemModule) RestoreSnapshot(ctx context.Context, className, snapshotID string) error {
-	if err := ctx.Err(); err != nil {
-		return errors.Wrapf(err, "restore snapshot aborted, invalid context")
-	}
-
-	metaPath := m.makeMetaFilePath(snapshotID)
-
-	metaData, err := os.ReadFile(metaPath)
+func (m *StorageFileSystemModule) RestoreSnapshot(ctx context.Context, className, snapshotId string) error {
+	snapshot, err := m.loadSnapShotMeta(ctx, className, snapshotId)
 	if err != nil {
-		return errors.Wrapf(err, "Could not read snapshot meta file %v", metaPath)
-	}
-	var snapshot snapshots.Snapshot
-	if err := json.Unmarshal(metaData, &snapshot); err != nil {
-		return errors.Wrapf(err, "Could not unmarshal snapshot meta file %v", metaPath)
+		return errors.Wrapf(err, "load snapshot meta")
 	}
 
 	for _, srcRelPath := range snapshot.Files {
 		if err := ctx.Err(); err != nil {
 			return errors.Wrapf(err, "restore snapshot aborted, system might be in an invalid state")
 		}
-		if err := m.copyFile(m.dataPath, m.makeSnapshotDirPath(snapshotID), srcRelPath); err != nil {
+		if err := m.copyFile(m.dataPath, m.makeSnapshotDirPath(className, snapshotId), srcRelPath); err != nil {
 			return errors.Wrapf(err, "restore snapshot aborted, system might be in an invalid state: file %v", srcRelPath)
 		}
 	}
 	return nil
 }
 
-func (m *StorageFileSystemModule) SetMetaStatus(ctx context.Context, className, snapshotID, status string) error {
-	// TODO implement
-	return nil
+func (m *StorageFileSystemModule) loadSnapShotMeta(ctx context.Context, className, snapshotID string) (*snapshots.Snapshot, error) {
+	if err := ctx.Err(); err != nil {
+		return nil, errors.Wrapf(err, "restore snapshot aborted, invalid context")
```

Comment:
This error message should change, because `loadSnapshotMeta` is not only called by Restore, but also by MetaStatus.

Also, the `ctx.Err()` already specifies that the context is invalid, so this wrap message here is redundant. As mentioned above, ideally, this wrap message should just contain information about what *this* function does, "load snapshot meta" in this case.